includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -778,7 +778,7 @@ public function process_redirect_payment( $order_id, $intent_id, $save_payment_m
 				$this->set_payment_method_title_for_order( $order, $payment_method_type, $payment_method_details );
 				$this->order_service->attach_transaction_fee_to_order( $order, $charge );
 
-				self::remove_upe_payment_intent_from_session();
+				static::remove_upe_payment_intent_from_session();
```

Comment:
I think these should all be `static` calls and this was actually a dormant bug. When placed here, `self` will always call `UPE_Payment_Gateway::remove_upe_payment_intent_from_session()` instead of `UPE_Split_Payment_Gateway::remove_upe_payment_intent_from_session()` when the calling class is instead an instance of `UPE_Split_Payment_Gateway`.